mycli/sqlexecute.py

```diff
@@ -80,12 +98,33 @@ def connect(self, database=None, user=None, password=None, host=None,
             FIELD_TYPE.DATE: lambda obj: (convert_date(obj) or obj),
         })
 
-        conn = pymysql.connect(database=db, user=user, password=password,
-                host=host, port=port, unix_socket=socket,
-                use_unicode=True, charset=charset, autocommit=True,
-                client_flag=pymysql.constants.CLIENT.INTERACTIVE,
-                local_infile=local_infile,
-                conv=conv, ssl=ssl, program_name="mycli")
+        defer_connect = False
+
+        if ssh_host:
+            defer_connect = True
+
+        conn = pymysql.connect(
+            database=db, user=user, password=password, host=host, port=port,
+            unix_socket=socket, use_unicode=True, charset=charset,
+            autocommit=True, client_flag=pymysql.constants.CLIENT.INTERACTIVE,
+            local_infile=local_infile, conv=conv, ssl=ssl, program_name="mycli",
+            defer_connect=defer_connect
+        )
+
+        if ssh_host and paramiko:
+            if not ssh_user:
+                ssh_user = getpass.getuser()
```

Comment:
@meeuw It looks like paramiko will already call `getpass.getuser()` if the username is `None`. So, maybe we don't need this on the mycli side?

Reply:
thanks, you're right.